tests/test_forward_ref.py

```diff
@@ -644,6 +644,8 @@ class Model(BaseModel):
     )
 
     m = module.Model(foo_user={'x': 'user1'}, user={'y': 'user2'})
+    # TODO: Do we need to support models_as_dict=False?
+    #   If not, is there a way to replicate this custom-encoder functionality?
     assert m.model_dump_json(models_as_dict=False) == '{"foo_user": {"x": "user1"}, "user": "User(user2)"}'
```

Comment:
Question 4: Without something like `models_as_dict=False`, I don't think see a good way to customize the encoding for the models; is there a way to handle this in v2?

Reply:
Is the idea that you'd wrap the model with a root serializer through an annotation?

If you have to add the root serializer to the model class itself then you'd lose the ability to change the behavior for different parent classes / the root class itself, which I suspect would be a pain point for migration. But as long as you can change the behavior through an annotation then I think it should be okay.

Reply:
either an annotation, or a field serialiser - e.g. the `@serialiser` decorator.